### PR TITLE
fix(AutoLayout): Support primaryAxisAlignment with independent layout

### PR DESCRIPTION
**Pull request by Robert-Louis**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

- Bugfix
## What is the current behavior?

- AutoLayout independent layout was not working with PrimaryAxisAlignment center or end

## What is the new behavior?

- PrimaryAxisAlignment center or end behave as wanted with independant layout

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs [hunk 1]
```diff
@@ -203,9 +203,12 @@ private void UpdateAlignments()
 			}
 
 			var independentLayoutCount = Children.Where(child => GetIsIndependentLayout(child)).Count();
+			var hasIndependentLayout = independentLayoutCount > 0;
 
 			var gridDefinitionsCount = isSpaceBetween ? ((childrenCount - independentLayoutCount) * 2) - 1 : (childrenCount - independentLayoutCount);
 
+			var isPrimaryAxisAlignmentCenterOrEnd = hasIndependentLayout && (PrimaryAxisAlignment == AutoLayoutAlignment.Center || PrimaryAxisAlignment == AutoLayoutAlignment.End);
```
Comment (Youssef1313) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs:210]: nit

```suggestion
			var isPrimaryAxisAlignmentCenterOrEnd = hasIndependentLayout && PrimaryAxisAlignment is AutoLayoutAlignment.Center or AutoLayoutAlignment.End;
```

**Review by carldebilly — APPROVED** [review]

**Comment by Xiaoy312** [comment]
@carldebilly 
UpdateAlignments has two large symmetrical sections
could it benefit from an adapter on Grid that invoke the respective vertical/horizontal parts based on context?
maybe not in this same PR

**Comment by carldebilly** [comment]
> @carldebilly UpdateAlignments has two large symmetrical sections could it benefit from an adapter on Grid that invoke the respective vertical/horizontal parts based on context? maybe not in this same PR

This code is potentially in critical path, so any _useless_ operations will slow down applications using this control.

**Review by carldebilly — APPROVED** [review]

**Pull request merged**